src/Raven.Client/Documents/Operations/Operation.cs

```diff
@@ -246,6 +246,12 @@ public void OnNext(OperationStatusChange change)
                         break;
                     case OperationStatus.Faulted:
                         StopProcessing();
+                        if (_additionalTask.IsFaulted)
+                        {
+                            _result.TrySetException(_additionalTask.Exception.ExtractSingleInnerException());
```

Comment:
```suggestion
                            _result.TrySetException(_additionalTask.Exception);
```

`.ExtractSingleInnerException()` should be on the caller, not here. We don't want to possibly lose information for the caller

Reply:
Have you pushed this?

Reply:
yes